### PR TITLE
Refactor/forecast residuals fn

### LINKED ISSUE #1127: add past_covariates and future_covariates to `residuals()`
This should be quite trivial

### PR DESCRIPTION
**Pull request by eliane-maalouf**
Fixes #1127.

### Summary

- Added past/future covariates in forecast_model/residuals()
- added UnitTest for residuals (tests/utils/test_residuals.py) - moved an already existing function testing residuals from tests/models/forecasting/test_backtesting.py)

### Other Information

- Corrected documentation for x_64 environment emulation on Apple M1 processors in INSTALL.md and CONTRIBUTING.md

### CODE REVIEW

File: darts/models/forecasting/forecasting_model.py [hunk 1]
```diff
@@ -913,14 +919,31 @@ def residuals(
         TimeSeries
             The vector of residuals.
         """
-        series._assert_univariate()
+        try:
+            series._assert_univariate()
+        except (AttributeError, TypeError):
```
Comment (hrzn) [inline darts/models/forecasting/forecasting_model.py:924]: 👍

File: darts/tests/utils/test_residuals.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import numpy as np
```
Comment (hrzn) [inline darts/tests/utils/test_residuals.py:1]: +1 for the separate test file

File: darts/tests/utils/test_residuals.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        model_instance = LinearRegressionModel(
+            lags=4, lags_past_covariates=4, lags_future_covariates=(4, 1)
+        )
+        model_instance.fit(
+            series=target_series,
+            past_covariates=past_covariates,
+            future_covariates=future_covariates,
+        )
+        # it seems that models can be fit with data objects that are not necessarily TimeSeries, but residuals() will
```
Comment (hrzn) [inline darts/tests/utils/test_residuals.py:57]: The only other possible type should be `Sequence[TimeSeries]` 😅

File: darts/tests/utils/test_residuals.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        with self.assertRaises(ValueError):
+            model.residuals(target_series, future_covariates=future_covariates)
+
+        with self.assertRaises(ValueError):
+            model.residuals(
+                target_series,
+                past_covariates=past_covariates,
+                future_covariates=future_covariates,
+            )
```
Comment (hrzn) [inline darts/tests/utils/test_residuals.py:75]: Could you also add a test case where the residuals computation carries out successfully with past and/or future covariates?

**Review by hrzn — COMMENTED** [review]
Looks good 👍 I only added a suggestion to add one more unit test (testing the actual residual computations with covariates).

**Comment by hrzn** [comment]
Some of the unit tests fail, but this is because #1228

File: darts/tests/utils/test_residuals.py (hunk 4 above)
Comment (eliane-maalouf) [reply darts/tests/utils/test_residuals.py:75]: ok will do

**Review by hrzn — APPROVED** [review]
LGTM, nice :)

**Pull request merged**